From 73f8958bfce39a69819063cc788a8021f398520a Mon Sep 17 00:00:00 2001
From: Yazan Shhady <yazan.shhady@solid-run.com>
Date: Mon, 2 Dec 2024 14:57:22 +0200
Subject: [PATCH] imx8mp_solidrun: update SKU parsing to detect SRHBIIOTIVxx
 (HB-IIOT)

Enhanced the `board_id_from_tlv_info` function to include detection of the HB-IIOT board.
Added a check for the "HBI" identifier in the TLV part number array,
ensuring proper recognition of SRHBIIOTIVxx as HB-IIOT.
---
 board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c b/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c
index ac4d56b6dc..d4b675e248 100644
--- a/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c
+++ b/board/solidrun/imx8mp_solidrun/imx8mp_solidrun.c
@@ -473,7 +473,8 @@ static void board_id_from_tlv_info(void) {
 
 	for(int i = 0; i < TLV_MAX_DEVICES; i++) {
 		// parse sku - processor or carrier indicated at index 2-6
-		if(memcmp(&hb_tlv_data.tlv_part_number[i][2], "HBC", 3) == 0) {
+		if(memcmp(&hb_tlv_data.tlv_part_number[i][2], "HBC", 3) == 0 ||
+			memcmp(&hb_tlv_data.tlv_part_number[i][2], "HBI", 3) == 0 ) {
 			/*
 			HummingBoard:
 				SKU - Board_Name {xx: board version}:
@@ -482,6 +483,7 @@ static void board_id_from_tlv_info(void) {
 				SRHBCUPRO0IVxx  HB-Pro
 				SRHBCME000CVxx  HB-Mate
 				SRHBCRE000CVxx  HB-Ripple
+				SRHBIIOTIVxx  	HB-IIOT
 			*/
 			switch(hb_tlv_data.tlv_part_number[i][5]) {
 			    case 'M': // Mate
-- 
2.25.1

